userfiles/fsl_melodic_output/fsl_melodic_output.rb

```diff
@@ -21,4 +21,162 @@
 
 class FslMelodicOutput < FileCollection
   Revision_info=CbrainFileRevision[__FILE__] #:nodoc:
+
+  has_viewer :name => "Melodic Viewer",  :partial => :melodic_viewer
+
+  ##################################################################################################
+  # This method returns the content of an HTML file modified so that it can be
+  # displayed in the content section of the userfile page.  It
+  # basically removes some HTML tags (html, body, object, head, title)
+  # and tweaks the URLs found in <a> and <img> tags.  It is quite
+  # specific to the HTML pages generated by FSL melodic. A more
+  # generic implementation would require a complete parsing of the
+  # HTML code.  It is recursively called for each iframe contained in
+  # the page.
+  # Parameters:
+  #  * file_path: the path of the HTML file whose content will be read, modified and returned. 
+  #  * dir_name: the directory name of the HTML file in the melodic file collection.
+  ##################################################################################################
+  
+  def modified_file_content file_path,dir_name
+    return nil unless File.exists?(file_path)
+    lines = Array.new
+    # The file is processed line by line to speed up substitutions.
+    File.open(file_path).each do |line|
+      # Tweaks hrefs.
+      new_line = tweak_hrefs_in_line line,dir_name
+      # Tweaks imgs.
+      new_line = tweak_imgs_in_line new_line,"#{self.name}/#{dir_name}"
+      # Remove specific tags.
+      new_line = new_line.gsub(/<object.*object>/i,"")
+      new_line = new_line.gsub(/<link.*stylesheet.*>/i,"")
+      new_line = new_line.gsub(/<html>/i,"")
+      new_line = new_line.gsub(/<\/html>/i,"")
+      new_line = new_line.gsub(/<body>/i,"")
+      new_line = new_line.gsub(/<\/body>/i,"")
+      new_line = new_line.gsub(/<head>/i,"")
+      new_line = new_line.gsub(/<\/head>/i,"")
+      new_line = new_line.gsub(/<title>.*<\/title>/i,"")
+      # Handles iframes.
+      keep_line = true
+      new_line.scan(/<IFRAME.*src=(.*) /i) do |x,y|
+	frame_path = File.dirname(file_path)+"/"+x
+   	lines << modified_file_content(frame_path,dir_name)
+	keep_line = false					  
+      end
+      lines << new_line if keep_line
+    end
+    return lines
+  end
+
+  ##################
+  # Helper methods #
+  ##################
+  
+  private
+
+  # Returns the min of two values.
```

Comment:
Yes, except that a.min raises an exception when an element of a is nil. See discussion above.